### PR TITLE
apfree-wifidog: Update to v3.11.1716

### PR DESCRIPTION
**Pull request by liudf0716**
Maintainer: Dengfeng Liu / @liudf0716
Compile tested: i386_pentium4, VMware® Workstation, master
Run tested: VMware® Workstation

Change log for v3.11.1716:
[IMP] Replaced libhttpd with libevent
[IMP] Added REQUEST_TYPE_COUNTERS_V2 to wifidog protocol
[IMP] Sent online and offline client's counter info to auth server

[FIX] Fixed missing setting online_time parameter bug

Signed-off-by: Dengfeng Liu <liudf0716@gmail.com>

### CODE REVIEW

**Comment by neheb** [comment]
The commit message is lacking the first line of this pull request.

**Comment by liudf0716** [comment]
@neheb fixed

File: net/apfree-wifidog/Makefile [hunk 1]
```diff
@@ -8,8 +8,8 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=apfree-wifidog
-PKG_VERSION:=3.8.1588
-PKG_RELEASE=1
+PKG_VERSION:=3.11.1716
+PKG_RELEASE=2
```
Comment (yousong) [inline net/apfree-wifidog/Makefile]: `PKG_RELEASE` should be reset to 1 on `PKG_VERSION` bump.

File: net/apfree-wifidog/Makefile [hunk 2]
```diff
@@ -31,26 +29,22 @@ define Package/apfree-wifidog
...
 define Package/apfree-wifidog/description
-  The ApFree Wifidog project is a complete and embeddable captive portal
-  solution for wireless community groups or individuals who wish to open a free
-  Hotspot while still preventing abuse of their Internet connection.
-
-  It's enhanced wifidog
+	The ApFree Wifidog project is a complete and embeddable captive portal
+	solution for wireless community groups or individuals who wish to open a free
+	Hotspot while still preventing abuse of their Internet connection.
+	It's enhanced wifidog
```
Comment (yousong) [inline net/apfree-wifidog/Makefile]: It's a convention when writing Makefile that we only use tabs for recipes.

File: net/apfree-wifidog/files/wifidogx.init [hunk 3]
```diff
@@ -98,7 +97,10 @@ prepare_wifidog_conf() {
 	[ -f ${CONFIGFILE} ] && rm -f ${CONFIGFILE}
 
 	config_get enable "${cfg}" "enable" 0
-	[ "${enable}" = "1" ] || return
+	[ "${enable}" = "1" ] || {
+		echo "wifidog disabled in /etc/config/wifidogx file, please set it to 1 to enable it"
```
Comment (yousong) [inline net/apfree-wifidog/files/wifidogx.init]: log should go to stderr `>&2`.

File: net/apfree-wifidog/files/wifidogx.conf [hunk 4]
```diff
@@ -1,13 +1,9 @@
...
-	option	auth_server_path	'/wifidog/'
-	option	check_interval		60
-	option	client_timeout		5
-	option	httpd_max_conn		200
-	option	pool_mode			1
-	option	thread_number		5
-	option	queue_size			20
-	option	wired_passed		0
-	option	enable		0
+config wifidogx
```
Comment (yousong) [inline net/apfree-wifidog/files/wifidogx.conf]: Please be aware that this section type name change will break compatibility and users' old config will be rendered useless.  OpenWrt tries hard to not break users' existing config even in master branch.

Comment (liudf0716) [reply net/apfree-wifidog/files/wifidogx.conf]: @yousong yes I agree with u, there are 2 reason why I change it
1, apfree wifidog was received by openwrt a few days ago, there are no history problems
2, mixed use wifidog and wifidogx will confuse users
that's why I changed it

Comment (liudf0716) [reply net/apfree-wifidog/files/wifidogx.conf]: restore it

Comment (yousong) [reply net/apfree-wifidog/files/wifidogx.conf]: It's not that such changes are forbidden, just that we need to be more careful when making changes that can cause behavior changes for existing users.  If you really intend to make such changes, it's also okay, but be sure to present the reasons to back it.

I noticed that are also uci options dropped in this pull request.  These also deserve a note in the commit message.  Thank you.

Comment (liudf0716) [reply net/apfree-wifidog/files/wifidogx.conf]: @yousong as u see, this version has replaced libhttpd with libevent, there were very big change. for new version thread pool mode was deprecated

**Comment by liudf0716** [comment]
@yousong @neheb  thanks

**Pull request merged**